web-examples/src/main/java/io/vertx/example/web/mongo/Routes.java

```diff
@@ -0,0 +1,30 @@
+package io.vertx.example.web.mongo;
+
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.ext.web.templ.JadeTemplateEngine;
+
+/**
+ * Handler for index page.
+ *
+ * @author <a href="mailto:pmlopes@gmail.com>Paulo Lopes</a>
+ */
+public class Routes {
+
+  // In order to use a JADE template we first need to create an engine
+  final static JadeTemplateEngine jade = JadeTemplateEngine.create();
+
+  public static void index(RoutingContext ctx) {
+    // we define a hardcoded title for our application
+    ctx.put("title", "Vert.x Web");
+
+    // and now delegate to the engine to render it.
+    jade.render(ctx, "templates/index", res -> {
```

Comment:
the issue with TemplateHandler is that it assumes the URL path as the file name, which would mean that i would create a template name `.jade`. Also the TemplateHandler does not allow me to reuse the template (again same reason, path must match template filename).

I think we should modify the TemplateHandler to have an extra arg, and if that extra arg is defined, then it will be always the filename, otherwise fallback to current implementation.